lib/Doctrine/Annotations/Metadata/PropertyMetadata.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Annotations\Metadata;
+
+/**
+ * @internal
+ */
+final class PropertyMetadata
+{
+    /** @var string */
+    private $name;
+
+    /** @var array<string, string>|null */
+    private $type;
+
+    /** @var bool */
+    private $required;
+
+    /** @var bool */
+    private $default;
+
+    /** @var mixed[]|null */
+    private $enum;
+
+    /**
+     * @param array<string, string> $type
+     * @param mixed[]|null          $enum
```

Comment:
Is it possible to have a stricter type declaration here? Specifically, do we allow objects and resources? I think we only allow constant-compatible values here.